eventuals/interrupt.h

```diff
@@ -63,22 +129,59 @@ class Interrupt final {
       return false;
     }
 
+    Handler::State state = handler->state_.state();
+    CHECK_EQ(state, Handler::State::UNINSTALLED);
+
     bool installed = handler_.compare_exchange_weak(
         stored,
         handler,
         std::memory_order_release,
         std::memory_order_relaxed);
 
-    // 'stored' might be a 'nullptr' only in case of race betweer
-    // 'load' and 'compare_exchange_weak'.
+    // 'stored' is 'nullptr' only when race between 'load' and
+    // 'compare_exchange_weak' occurs, so 'stored' is not stored
+    // in the 'handler_'.
     CHECK(installed || stored == nullptr);
+
+    CHECK(handler->state_.Update(state, Handler::State::INSTALLED));
+
+    if (stored != nullptr) {
+      Handler::State state = stored->state_.state();
+      while (!stored->state_.Update(state, Handler::State::UNINSTALLED)) {}
+    }
+
     return installed;
   }
 
+  [[nodiscard]] bool Uninstall(Handler* handler) {
+    // If 'handler' is current 'handler_' then the interrupt *must not*
+    // have been triggered so we should set it back to 'placeholder_handler_'
+    // instead of nullptr. If we successfully did the compare-and-swap then
```

Comment:
```suggestion
    // If 'handler' is the current 'handler_' then the interrupt *must not*
    // have been triggered so we want to set 'handler_' back to 'placeholder_handler_'
    // instead of nullptr. If we successfully did the compare-and-swap then that must mean
    // that 'handler' is the current 'handler_' so we "won" the race of uninstalling and
```